pkg/controllers/status/cluster_status_controller.go

```diff
@@ -361,11 +377,16 @@ func getKubernetesVersion(clusterClient *util.ClusterClient) (string, error) {
 	return clusterVersion.GitVersion, nil
 }
 
+// getAPIEnablements get the list of APIs installed in the member cluster
 func getAPIEnablements(clusterClient *util.ClusterClient) ([]clusterv1alpha1.APIEnablement, error) {
+	// This can return an error *and* the results it was able to find.
+	// If the length of api resources is not zero,we don't need to fail on the error.
 	_, apiResourceList, err := clusterClient.KubeClient.Discovery().ServerGroupsAndResources()
-	if err != nil {
+	if len(apiResourceList) == 0 {
+		klog.Errorf("unable to get any supported resources from server. Error: %v.", err)
 		return nil, err
 	}
```

Comment:
Can you split a PR for the partially APIs? This is our key mission, and shouldn't be blocked by the optional new condition.

Reply:
> Can you split a PR for the partially APIs?

I am not clearly  about what you sayed.Do you mean this pr should not contain e2e test,just focus on the feature of  `metadataSufficient `status.Or is there a problem in line 385 and 386 ?

Reply:
> Do you mean this pr should not contain e2e test,just focus on the feature of `metadataSufficient `status

If my understand is right,I will split this PR in tree:1. Add metadataSufficient  status 2. Add output of metadataSufficient  in `kubectl get cluster` 3.Add new e2e test for metadataSufficient  status

Reply:
> Can you split a PR for the partially APIs? This is our key mission, and shouldn't be blocked by the optional new condition.

If you do not mind,Can you describe it in Chinese?

Reply:
I mean the change for `func getAPIEnablements` is what we urgently needed. We can do it in a separate PR.

For the `MetadataSufficient` condition, I'm a little bit hesitant as it seems to lack enough benefits to users.

Reply:
Thanks. I got it

Reply:
A separate PR in https://github.com/karmada-io/karmada/pull/1968 
/cc @RainbowMango